src/main/kotlin/org/rust/ide/annotator/RsAttrErrorAnnotator.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator
+
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.psi.PsiElement
+import org.rust.ide.inspections.fixes.SubstituteTextFix
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.elementType
+import org.rust.lang.utils.RsDiagnostic
+import org.rust.lang.utils.addToHolder
+
+class RsAttrErrorAnnotator : AnnotatorBase() {
+    override fun annotateInternal(element: PsiElement, holder: AnnotationHolder) {
+        if (element !is RsMetaItem) return
+        checkMetaBadDelim(element, holder)
+        checkAttrTemplateCompatible(element, holder)
+        checkLiteralSuffix(element, holder)
+    }
+}
+
+private fun checkLiteralSuffix(metaItem: RsMetaItem, holder: AnnotationHolder) {
+    val exprList = metaItem.metaItemArgs?.litExprList ?: return
+    for (expr in exprList) {
+        val kind = expr.kind
+        if (kind is RsLiteralWithSuffix) {
+            val suffix = kind.suffix ?: continue
+            val editedText = expr.text.removeSuffix(suffix)
+            val fix = SubstituteTextFix.replace(
+                "Remove suffix", metaItem.containingFile, expr.textRange, editedText
+            )
+            RsDiagnostic.AttributeSuffixedLiteral(expr, fix).addToHolder(holder)
+        }
+    }
+}
+
+private fun checkAttrTemplateCompatible(metaItem: RsMetaItem, holder: AnnotationHolder) {
+    val name = (metaItem.path ?: return).text
```

Comment:
```suggestion
    val name = metaItem.name ?: return
```

The extension property `name` returns non-null value only in the case of simple path, like `foo` but not `foo::bar`